### PR TITLE
r/aws_globalaccelerator_accelerator: Add aws_globalaccelerator_accelerator resource

### PR DESCRIPTION
**Pull request by gazoakley**
Partly addresses #6739

Changes proposed in this pull request:

* **New Resource:** `aws_globalaccelerator_accelerator`

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAwsGlobalAcceleratorAccelerator'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccAwsGlobalAcceleratorAccelerator -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_basic
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_basic
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_update
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_update
=== RUN   TestAccAwsGlobalAcceleratorAccelerator_attributes
=== PAUSE TestAccAwsGlobalAcceleratorAccelerator_attributes
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_basic
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_attributes
=== CONT  TestAccAwsGlobalAcceleratorAccelerator_update
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_basic (40.31s)
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_update (64.26s)
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_attributes (66.62s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	66.665s
```

### CODE REVIEW

File: aws/resource_aws_globalaccelerator_accelerator.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+		Create: resourceAwsGlobalAcceleratorAcceleratorCreate,
+		Read:   resourceAwsGlobalAcceleratorAcceleratorRead,
+		Update: resourceAwsGlobalAcceleratorAcceleratorUpdate,
+		Delete: resourceAwsGlobalAcceleratorAcceleratorDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
```
[the remark this answers is not in the document]
Comment (gazoakley) [reply aws/resource_aws_globalaccelerator_accelerator.go]: I thought it was good practise to offer customisable timeouts if code needs to do a wait, but I can't imagine users would want to change the defaults very often. Would it be better to add documentation or remove the ability to customize timeouts?

[the remark this answers is not in the document]
Comment (bflad) [reply aws/resource_aws_globalaccelerator_accelerator.go]: We generally want to err on the side of not providing it unless its dependent on some scalable resource (e.g. amount of data in a database) or fluctuates based on AWS-background processes (e.g. varying ENI removal time from load balancers when trying remove subnets). We can always add it later if necessary.

**Comment by gazoakley** [comment]
Happy New Year @bflad 🎉 - I'll make the updates shortly. One thing I meant to ask about was what would be the best way to deal with deletion when an accelerator is enabled - the call to `DeleteAccelerator` fails if `enabled = true`. I could:

* Automatically call `UpdateAccelerator` to disable the accelerator, then call `DeleteAccelerator`
* Conditionally do the above with some kind of force flag
* Call `DeleteAccelerator`, catch the "accelerator is enabled" error and retry after disabling the accelerator

Any preference?

**Comment by bflad** [comment]
Interesting and happy new year as well! Got to love how none of these services are consistent. 😂 First or third option sounds good, whatever is easiest.

File: aws/resource_aws_globalaccelerator_accelerator_test.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGlobalAcceleratorAcceleratorDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlobalAcceleratorAccelerator_basic(rName, true),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGlobalAcceleratorAcceleratorExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", newName),
```
Comment (bflad) [inline aws/resource_aws_globalaccelerator_accelerator_test.go]: Should be `rName` 😄 

```
--- FAIL: TestAccAwsGlobalAcceleratorAccelerator_update (35.05s)
    testing.go:538: Step 0 error: Check failed: Check 2/3 error: aws_globalaccelerator_accelerator.example: Attribute 'name' expected "tf-acc-test-1060790186692934857", got "tf-acc-test-4121192575072923489"
```

```suggestion
					resource.TestCheckResourceAttr(resourceName, "name", rName),
```

**Comment by bflad** [comment]
Whoops didn't post my approval first 😅 Thanks, @gazoakley! 🚀 

```
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_basic (55.12s)
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_attributes (69.74s)
--- PASS: TestAccAwsGlobalAcceleratorAccelerator_update (89.72s)
```

**Comment by bflad** [comment]
This has been released in [version 1.56.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1560-january-16-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**